checksec.cpp

```diff
@@ -63,6 +64,35 @@ Checksec::Checksec(std::string filepath) : filepath_(filepath), loadedImage_(fil
         loadConfigSecurityCookie_ = loadConfig.SecurityCookie;
         loadConfigSEHandlerTable_ = loadConfig.SEHandlerTable;
         loadConfigSEHandlerCount_ = loadConfig.SEHandlerCount;
+
+        // Iterate over debug directories
+        if (!peparse::GetDataDirectoryEntry(loadedImage_.get(), peparse::DIR_DEBUG,
+                                            debugDirectories)) {
+            std::cerr << "Warn: No debug directories"
+                      << "\n";
+            return;
+        }
+
+        peparse::debug_dir_entry debugDir{};
+        uint32_t numberOfDebugDirs = debugDirectories.size() / sizeof(debugDir);
+        auto debugDirSize = std::min(debugDirectories.size(), sizeof(debugDir));
+
+        for (int i = 0; i < numberOfDebugDirs; i++) {
+            // copy the current debug directory into debugDir
+            memcpy(&debugDir, debugDirectories.data() + (i * debugDirSize), debugDirSize);
+            // 20 == IMAGE_DEBUG_TYPE_EX_DLLCHARACTERISTICS
+            // For now we only care about this debug directory since it contains the CETCOMPAT bit
+            if (debugDir.Type == 20) {
```

Comment:
Can we use the `IMAGE_DLLCHARACTERISTICS_EX_CET_COMPAT` constant from `pe-parse` here?

Reply:
Looks like I never defined it as a constant in `pe-parse` :(